packages/blockchain-extension/extension/commands/submitTransaction.ts

```diff
@@ -89,61 +100,110 @@ export async function submitTransaction(evaluate: boolean, treeItem?: Instantiat
         }
     }
 
-    let argsString: string;
-    if (transactionObject) {
-        argsString = transactionObject.args;
-    } else {
-        argsString = await UserInputUtil.showInputBox('optional: What are the arguments to the transaction, (e.g. ["arg1", "arg2"])', '[]');
+    gatewayRegistryEntry = FabricGatewayConnectionManager.instance().getGatewayRegistryEntry();
+    let associatedTestData: {chaincodeName: string, testDataPath: string};
+    if (gatewayRegistryEntry.testDataDirectories) {
+        associatedTestData = gatewayRegistryEntry.testDataDirectories.find((item: {chaincodeName: string, testDataPath: string}) => {
+            return item.chaincodeName === smartContract;
+        });
     }
 
-    if (argsString === undefined) {
-        return;
-    } else if (argsString === '') {
-        args = [];
-    } else {
-        argsString = argsString.trim();
-        try {
-            if (!argsString.startsWith('[') || !argsString.endsWith(']')) {
-                throw new Error('transaction arguments should be in the format ["arg1", {"key" : "value"}]');
+    if (associatedTestData) {
+        const filepaths: string[] = [];
+        const userTransactions: ITransactionData[] = [];
+        const testDataDirPath: string = associatedTestData.testDataPath;
+        const quickPickItems: string[] = [];
+
+        fs.readdirSync(testDataDirPath).forEach((file: string) => {
+            if (file.endsWith('.json')) {
+                filepaths.push(file);
+            }
+        });
+
+        if (filepaths.length > 0) {
+            for (const file of filepaths) {
+                const fileContents: Buffer = await fs.readFile((path.join(testDataDirPath, file)));
+                fileJson = JSON.parse(fileContents.toString());
```

Comment:
what if the file can't be parsed you probably don't want it to not parse the rest. But you would need to inform the user